Week8/MovieViewer/app/src/main/java/com/wmw/movieviewer/ui/DetailActivity.kt

```diff
@@ -0,0 +1,43 @@
+package com.wmw.movieviewer.ui
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.lifecycleScope
+import com.bumptech.glide.Glide
+import com.wmw.movieviewer.App
+import com.wmw.movieviewer.R
+import com.wmw.movieviewer.model.Movie
+import com.wmw.movieviewer.repository.MovieRepository
+import kotlinx.android.synthetic.main.activity_detail.*
+import kotlinx.coroutines.launch
+
+private const val MOVIE_KEY = "movie_id"
+
+fun startDetailActivity(from: Context, movieId: String?) =
+    from.startActivity(Intent(from, DetailActivity::class.java).apply {
+        putExtra(MOVIE_KEY, movieId)
+    })
+
+class DetailActivity : AppCompatActivity() {
+
+    private val movieId by lazy { intent.getStringExtra(MOVIE_KEY) }
+    private val repository by lazy { MovieRepository(App.movieDb.movieDao(), App.remoteApi) }
```

Comment:
Just like you fetched the MovieDao and RemoteApi through App., you could've fetched the repository too!